src/util/html.ts

```diff
@@ -1,39 +1,111 @@
 import * as sax from 'sax';
 import { Link } from '../link';
+import { resolve } from './uri';
 
-export function parseHtmlLinks(contextUri: string, body: string): Link[] {
+export type HtmlForm = {
+  action: string,
+  method: string | null,
+  enctype: string | null,
+  rel: string | null,
+  id: string | null,
+}
+
+type ParseHtmlResult = {
+
+  links: Link[],
+  forms: HtmlForm[],
+
+}
+
+export function parseHtml(contextUri: string, body: string): ParseHtmlResult {
 
   const parser = sax.parser(false, {});
   const links: Link[] = [];
+  const forms: ParseHtmlResult['forms'] = [];
 
   parser.onopentag = node => {
 
-    if (!node.attributes.REL) {
-      return;
-    }
-    if (!node.attributes.HREF) {
-      return;
+    switch(node.name) {
+      case 'LINK' :
+      case 'A' :
+        links.push(...parseLink(contextUri, node as sax.Tag));
+        break;
+      case 'FORM' :
+        forms.push(...parseForm(contextUri, node as sax.Tag));
+        break;
+
     }
 
-    const rels = <string> node.attributes.REL;
+  };
 
-    for (const rel of rels.split(' ')) {
+  parser.write(body).close();
 
-      const type = <string>node.attributes.TYPE;
-      const link: Link = {
-        rel,
-        context: contextUri,
-        href: <string> node.attributes.HREF,
-      };
-      if (type) link.type = type;
-      links.push(link);
+  return {
+    forms,
+    links,
+  };
 
-    }
+}
 
-  };
+function parseLink(contextUri: string, node: sax.Tag): Link[] {
 
-  parser.write(body).close();
+  if (!node.attributes.REL) {
+    return [];
+  }
+  if (!node.attributes.HREF) {
+    return [];
+  }
+
+  const rels = <string> node.attributes.REL;
+
+  const links: Link[] = [];
+  for (const rel of rels.split(' ')) {
+
+    const type = <string>node.attributes.TYPE;
+    const link: Link = {
+      rel,
+      context: contextUri,
+      href: <string> node.attributes.HREF,
+    };
+    if (type) link.type = type;
+    links.push(link);
+  }
 
   return links;
 
 }
+
+function parseForm(contextUri: string, node: sax.Tag): HtmlForm[] {
+
+  const rels = node.attributes.REL || null;
+  const id = node.attributes.ID || null;
```

Comment:
just wondering, is ID for this capitalized?

Reply:
ah, nvm, everything else in attributes are capitalized..

Reply:
Yea that's how the `sax` library encodes attribute and tag names

Reply:
good eye though